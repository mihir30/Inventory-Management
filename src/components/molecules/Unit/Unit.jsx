import React from 'react'
import BuildOutlinedIcon from '@mui/icons-material/BuildOutlined';
import "./Unit.scss"
const Unit = ({ status, serial, type, name, person, fromDate, id }) => {

  return (
    <>
      <tr className="unit-row units-table-tr table-bg-color">
        <td className="col-2 units-table-td"><i className='wrench-icon'><BuildOutlinedIcon fontSize='small' /></i>
          <span className={`badge rounded-pill bg-${status}`}>{status}</span></td>
        <td className="col-2 align-right units-table-td">{serial}</td>
        <td className="col-2 align-right units-table-td">{type}</td>
        <td className="col-2 align-right units-table-td">{name}</td>
        <td className="col-2 units-table-td ">{person}</td>
        <td className="col-2 units-table-td">{fromDate}</td>
      </tr>
    </>);
};

export default Unit
