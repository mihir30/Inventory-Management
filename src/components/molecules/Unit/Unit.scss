.toolbar-buttons {
  display: flex;
  justify-content: space-between;
  color: gray;
  align-items: center;
}
.table-bg-color{
  background-color: white;
}
.units-toolbar {
  background-color: #fff;
  border: 0.01rem solid lightgrey;
  display: flex;
  justify-content: space-between;
  margin: 1rem 0.8rem 0 0.8rem;
}
.toolbar-icon {
  margin: 0 0.6rem;
  cursor: pointer;
  padding: 0.1rem;
  width: 2rem;
  height: 2rem;
  text-align: center;
}
.toolbar-icon:hover {
  background-color: lightgrey;
  border-radius: 100%;
  transform: scale(0.9);
  transition: 1ms;
}
.all-units-text {
  color: gray;
  font-size: 16px;
  font-weight: bold;
  letter-spacing: 0.2px;
  margin: 10px;
}

.units-table-tbody {
  overflow-y: auto;
  width: 100%;
}

.units-table-thead,
.units-table-tbody,
.units-table-tr,
.units-table-td,
.units-table-th {
  display: block;
  font-size:14px;
}
.wrench-icon {
  margin-right: 1rem;
  // padding: 0 1rem;
  color: grey;
}
.units-table-tbody,
.units-table-td {
  display: inline-block;
}


.units-table-tbody ,
.units-table-th,
.units-table-thead > tr > th {
  float: left;
  position: relative;
  background-color: #ebedf0;
}

.unit-row {
  border-bottom: 1px solid #edeff0;
}

.unit-row-top {
  border-top: 1px solid #1ca1c1;
}

.bg-Available {
  background-color: #55cd97;
  font-size: 11px;
}

.bg-Taken {
  background-color: #1ca1c1;
  font-size: 11px;
}
.bg-Broken {
  background-color: #ff5c4c;
  font-size: 11px;
}

.units-table-thead {
  border-right: 0.01rem solid lightgrey;
  border-left: 0.01rem solid lightgrey;
  background-color: #ebedf0;
}

.search-box {
  display: contents;
}

.search-input {
  border: none;
  background: none;
  outline: none;
  float: left;
  padding: 0;

  font-size: 20px;
  transition: 0.4s;
  line-height: 50px;
  width: 0;
}

.search-box:hover > .search-input {
  width: 250px;
  padding: 0 7px;
  color: black;
}

.spinner-alignment{
  position: absolute;
  top: 50%;
  left: 50%;
}