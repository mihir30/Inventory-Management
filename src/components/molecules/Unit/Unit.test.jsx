import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Unit from './Unit';

describe("Units by State content section Component", () => {

    it('renders the component properly', () => {
        const trEl = document.createElement("tr");
        ReactDom.render(<BrowserRouter><Unit 
            status="Available" serial="PC-00017" type="Computer" name="Dell X2" person="Araceli" fromDate="2022-05-31"/></BrowserRouter>, trEl); 
        
    });

    it('should have the right units by state content section component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});