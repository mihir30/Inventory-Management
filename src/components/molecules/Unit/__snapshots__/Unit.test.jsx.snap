// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Units by State content section Component should have the right units by state content section component snapshot 1`] = `null`;
